trieste/models/gpflow/models.py

```diff
@@ -211,6 +232,183 @@ def evaluate_loss_of_model_parameters() -> tf.Tensor:
 
         multiple_assign(self.model, current_best_parameters)
 
+    def conditional_predict_f(
+        self, query_points: TensorType, additional_data: Dataset
+    ) -> tuple[TensorType, TensorType]:
+        """
+        Returns the marginal GP distribution at query_points conditioned on both the model
+        and some additional data, using exact formula.
+
+        :param query_points: Set of query points with shape [M, D]
+        :param additional_data: Dataset with query_points with shape [..., N, D] and observations
+                 with shape [..., N, L]
+        :return: mean_new: predictive variance at query_points, with shape [..., M, L],
+                 and var_new: predictive variance at query_points, with shape [..., M, L]
+        """
+
+        tf.debugging.assert_shapes(
+            [
+                (additional_data.query_points, [..., "N", "D"]),
+                (additional_data.observations, [..., "N", "L"]),
+                (query_points, ["M", "D"]),
+            ],
+            message="additional_data must have query_points with shape [..., N, D]"
+            " and observations with shape [..., N, L], and query_points "
+            "should have shape [M, D]",
+        )
+
+        if isinstance(self.model, SGPR):
+            raise NotImplementedError("Conditional predict f is not supported for SGPR.")
+
+        mean_old, cov_old = self.model.predict_f(
```

Comment:
for the sake of following equations more easily `mean_old` here is probably better named `mean_new_old`, since it corresponds to \hat{Z}(x_new)_old, right?

Reply:
actually the covariance is \Sigma_new?

Reply:
changed the names with add and qp, it's probably much clearer